### PR TITLE
Replaced some array return types

### PR DESCRIPTION
**Pull request by oscargus**

### CODE REVIEW

File: src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java [hunk 1]
```diff
@@ -315,11 +315,9 @@ private boolean handleDraggedFilenames(String s, final int dropRow) {
      * @return success status for the operation
      */
     private boolean handleDraggedFiles(List<File> files, final int dropRow) {
-        final String[] fileNames = new String[files.size()];
-        int i = 0;
+        final List<String> fileNames = new ArrayList<>();
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java]: The loop could be replaced with a nice stream/lamda expression:
`fileNames = new ArrayList<>(files.stream().map(f->f.getAbsolutePath()).collect(Collectors.toList());`

**Comment by Siedlerchr** [comment]
Except for the remark LGTM :+1:

File: src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java]: Yes, but I know that some developers not always appreciate these as they are harder to read. Have not really understood when they are preferred.

Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java]: I do not think they are harder to read, it is just some novel style, once you get used to it, it is plain simple and makes programming for fluent. In general I would prefer them if it makes sense.
The functional style makes it easier to crate a kind of  Collection-pipelline, e.g. introduce a filter, another transformation or limit the results or just replace the collector.  And could be easy parallelized, in cases where needed.

Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java]: I do not have any strong opinions but I know I made a PR a number of months
ago where I used these a lot and there were some objections.

Comment (simonharrer) [inline src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java]: I think we do not have a full consensus on this. But I would agree to @Siedlerchr in that case. :)

**Pull request merged**